### PR TITLE
feat(KONFLUX-6986): OLM bundle images should not be multi-arch

### PR DESCRIPTION
**Pull request by arewm**
If OLM bundle images are multi-arch then the image pulled by the client will be interpolated to match the client's platform. This can result in inconsitent behaviors between architectures. Instead, a single image manifest should be created since there is no binary and therefore no architecture dependencies.

resolves: KONFLUX-6986

### CODE REVIEW

File: policy/release/olm/olm.rego [hunk 1]
```diff
@@ -318,6 +317,29 @@ deny contains result if {
...
+#   short_name: olm_bundle_multi_arch
+#   failure_msg: The %q bundle image is a multi-arch reference.
+#   solution: >-
+#     Rebuild your bundle image without creating an image index.
+#   collections:
+#   - redhat
+#   effective_on: 2025-5-01T00:00:00Z
+deny contains result if {
+	# Parse manifests from snapshot
+	some csv_manifest in _csv_manifests
```
Comment (simonbaird) [inline policy/release/olm/olm.rego:335]: I think you need this, since otherwise there's no other test making sure csv_manifest is not empty.

```suggestion
	count(csv_manifest) > 0
```

Comment (simonbaird) [reply policy/release/olm/olm.rego:335]: Actually that's wrong. Disregard!

Comment (simonbaird) [reply policy/release/olm/olm.rego:335]: The some implies it can't be empty.

**Review by simonbaird — APPROVED** [review]

**Review by nmars — APPROVED** [review]
LGTM
